swim/memberlist.go

```diff
@@ -250,11 +277,47 @@ func (m *memberlist) MakeFaulty(address string, incarnation int64) []Change {
 	return m.MakeChange(address, incarnation, Faulty)
 }
 
+// TODO: MakeLeave is only during testing. Need to figure out if we want to keep
+// it around or rewrite the tests.
 func (m *memberlist) MakeLeave(address string, incarnation int64) []Change {
+	// for backwards compatibility in tests we allow to update the local member
+	// in place when the gossip is about the local node and an incarnation number
+	// that is expected to have accept the update.
+	if address == m.local.Address && m.local.Incarnation <= incarnation {
+		m.local.Status = Leave
```

Comment:
Since this function is only being used in tests I will un-export the function and document that it is only used in tests. I'm even inclined to move it to the test file but I'm not completely sure about that.

Then in a subsequent PR I will try to remove all obsolete `Make` functions (eg. `MakeLeave` and `MakeAlive`) since only `MakeSuspect`, `MakeFaulty` and `MakeTombstone` are allowed to be used by any given node. All other states can only be declared by the node its self and therefore should be routed via `SetLocalStatus`.